### PR TITLE
fix(kms): Detect `sstable not found` situation better

### PR DESCRIPTION
**Pull request by vponomaryov**
Also, allow some part of the sstables to fail the encryption check because we care about big picture and not concrete number.

### Testing

- [x] [https://argus.scylladb.com/test/105e12f0-26ba-4d7d-b945-53e472772f3f/runs?additionalRuns[]=97f28117-6097-489a-93db-82787916131e](https://argus.scylladb.com/test/105e12f0-26ba-4d7d-b945-53e472772f3f/runs?additionalRuns%5B%5D=97f28117-6097-489a-93db-82787916131e)

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

**Comment by vponomaryov** [comment]
Example of the failure is here: https://argus.scylladb.com/test/0622c223-d2fb-4548-bb06-5f071009b714/runs?additionalRuns%5B%5D=810ec91c-dfe1-40ab-acbc-11d36e796e3a

Example of the fix is here: https://argus.scylladb.com/test/105e12f0-26ba-4d7d-b945-53e472772f3f/runs?additionalRuns[]=97f28117-6097-489a-93db-82787916131e

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -4094,9 +4094,7 @@ def upgrade_sstables(nodes):
 
         @retrying(n=4, sleep_time=30, allowed_exceptions=(AssertionError, ))
         def check_encryption_fact(sstable_util_instance, expected_bool_value):
-            encryption_results = sstable_util_instance.is_sstable_encrypted()
-            assert encryption_results
-            assert all(encryption_result is expected_bool_value for encryption_result in encryption_results)
+            sstable_util_instance.check_that_sstables_are_encrypted()
```
Comment (fruch) [inline sdcm/nemesis.py:4097]: now expected_bool_value is now used, you lost the capability of testing for clear...

Comment (vponomaryov) [reply sdcm/nemesis.py]: Fixed.

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**